src/coreclr/tools/aot/ILCompiler/ILCompiler.csproj

```diff
@@ -1,18 +1,21 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputPath>$(RuntimeBinDir)ilc/</OutputPath>
-    <RuntimeIdentifier>$(PackageRID)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(_IsPublishing)' != 'true'">$(PackageRID)</RuntimeIdentifier>
```

Comment:
Where is the `_IsPublishing` property set? 

@agocke's comment https://github.com/dotnet/runtime/pull/107772#issuecomment-2350008766 makes it sound like the `_IsPublishing` is not set as expected in this project.

Reply:
Right now the dotnet CLI sets `_IsPublishing` when running `dotnet publish` but it's everyone else's responsibility to set it (I don't like this behavior, fyi). As far as I can tell, ILCompiler doesn't guarantee this is set, currently.